p2p/src/sync/peer.rs

```diff
@@ -203,12 +209,36 @@ where
         Ok(())
     }
 
+    /// Delays the processing of a new block until it can be accepted by the chainstate (but not more than `max_clock_diff`).
+    /// This is needed to allow the local or remote node to have slightly inaccurate clocks.
+    /// Without it, even a 1 second difference can break block synchronization
+    /// because one side may see the new block as invalid.
+    async fn sync_clock(&self, block_timestamp: BlockTimestamp) {
+        let block_timestamp = block_timestamp.as_duration_since_epoch();
+        let max_block_timestamp =
+            self.time_getter.get_time() + *self.chain_config.max_future_block_time_offset();
+        if block_timestamp <= max_block_timestamp {
+            return;
+        }
+        let clock_diff = max_block_timestamp - block_timestamp;
+        let sleep_time = std::cmp::max(clock_diff, *self.p2p_config.max_clock_diff);
```

Comment:
Should this be `std::cmp::min` instead? According to the doc comment:
> Delays the processing of a new block ... (but not more than `max_clock_diff`).
In which case if the `clock_diff` is larger than `max_clock_diff`, we should sleep `max_clock_diff` instead of `clock_diff`.

Reply:
Nice catch, thanks!